x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_request_schema.test.ts

```diff
@@ -5,59 +5,48 @@
  * 2.0.
  */
 
-import * as t from 'io-ts';
-import { left } from 'fp-ts/lib/Either';
-import { pipe } from 'fp-ts/lib/pipeable';
-import { exactCheck, foldLeftRight, getPaths } from '@kbn/securitysolution-io-ts-utils';
-
+import { expectParseError, expectParseSuccess, stringifyZodError } from '@kbn/zod-helpers';
```

Comment:
I have submitted invalid request to rule, and error message does not give any hints, what's wrong

```
POST http://localhost:5601/kbn/api/detection_engine/rules
{
    "type": "esql",
    "language": "esql",
    "query": "from packetbeat-8.5.3 | stats counted = count_distinct(destination.domain)",
    "author": [],
    "false_positives": [],
    "references": [],
    "risk_score": 21,
    "risk_score_mapping": [],
    "severity": "low",
    "severity_mapping": [],
    "name": "sample rule",
    "description": "sample rule",
    "tags": [],
    "license": "",
    "interval": "45m",
    "from": "now-2760s",
    "to": "now",
    "meta": {
        "from": "1m",
        "kibana_siem_app_url": "http://localhost:5601/kbn/app/security"
    },
    "actions": [],
    "enabled": false
}
```

response

```
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "[request body]: Invalid input"
}
```
I am afraid that can make nearly impossible addressing any issues by customers or by us in SDHs or debugging/writing code

With this request and response, can you somehow determine what's wrong with request, unless going through the schema manually?

Reply:
Btw, this request is passing with old schema. It seems we've got some breaking change

Reply:
See https://github.com/elastic/kibana/pull/167999#discussion_r1371610782